metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java

```diff
@@ -0,0 +1,187 @@
+     * @param uris         uris
+     * @param prefixSearch prefixSearch
+     * @return partition names
+     */
+    @Override
+    public Map<String, List<QualifiedName>> getPartitionNames(
+        @Nonnull final ConnectorRequestContext context,
+        @Nonnull final List<String> uris,
+        final boolean prefixSearch) {
+        return directSqlGetPartition.getPartitionNames(context, uris, prefixSearch);
+    }
+
+    @Override
+    protected Map<String, PartitionHolder> getPartitionsByNames(final Table table, final List<String> partitionNames) {
+        return directSqlGetPartition.getPartitionHoldersByNames(table, partitionNames);
+    }
+
+    protected void addUpdateDropPartitions(final QualifiedName tableQName, final Table table,
+        final List<String> partitionNames, final List<PartitionInfo> addedPartitionInfos,
+        final List<PartitionHolder> existingPartitionHolders, final Set<String> deletePartitionNames) {
+        final boolean  useHiveFastServiceForSavePartitions = Boolean.parseBoolean(getContext().getConfiguration()
+                .getOrDefault("hive.use.embedded.fastservice.save.partitions", "false"))
+            || (table.getParameters() != null && Boolean.parseBoolean(table.getParameters()
+                .getOrDefault("hive.use.embedded.fastservice.save.partitions", "false")));
+        if (useHiveFastServiceForSavePartitions) {
+            if (!existingPartitionHolders.isEmpty()) {
+                copyTableSdToPartitionHoldersSd(existingPartitionHolders, table);
+            }
+            copyTableSdToPartitionInfosSd(addedPartitionInfos, table);
+            directSqlSavePartition.addUpdateDropPartitions(tableQName, table, addedPartitionInfos,
+                existingPartitionHolders, deletePartitionNames);
+        } else {
+            super.addUpdateDropPartitions(tableQName, table, partitionNames, addedPartitionInfos,
+                existingPartitionHolders, deletePartitionNames);
+        }
+    }
+
+    private void copyTableSdToPartitionInfosSd(final List<PartitionInfo> partitionInfos, final Table table) {
+        //
+        // Update the partition info based on that of the table.
+        //
+        for (PartitionInfo partitionInfo : partitionInfos) {
+            copyTableSdToPartitionInfoSd(partitionInfo, table);
+        }
+    }
+
+    private void copyTableSdToPartitionHoldersSd(final List<PartitionHolder> partitionHolders, final Table table) {
+        //
+        // Update the partition info based on that of the table.
+        //
+        for (PartitionHolder partitionHolder : partitionHolders) {
+            copyTableSdToPartitionInfoSd(partitionHolder.getPartitionInfo(), table);
+        }
+    }
+
+    private void copyTableSdToPartitionInfoSd(final PartitionInfo partitionInfo, final Table table) {
+        final StorageInfo sd = partitionInfo.getSerde();
+        final StorageDescriptor tableSd = table.getSd();
+
+        if (Strings.isNullOrEmpty(sd.getInputFormat())) {
```

Comment:
Do we care about blanks here? e.g. is `"     "` ok?  if not I would switch this to `isNotBlank`

Reply:
Will use isNotBlank.